client/src/konva_mindmap/hooks/useCollisionDetectionForWorker.ts

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useRef, useEffect, useState } from "react";
+import Konva from "konva";
+import { Node, NodeData } from "@/types/Node";
+
+export function useCollisionDetection(nodeData: NodeData, updateNode: (id: number, updates: Partial<Node>) => void) {
+  const [collisionWorker, setCollisionWorker] = useState<Worker | null>(null);
+  const layer = useRef<Konva.Layer>(null);
+
+  useEffect(() => {
+    const worker = new Worker(new URL("../utils/collision-worker.ts", import.meta.url));
+
+    worker.onmessage = (event) => {
+      if (event.data.type === "collisionResult") {
+        event.data.collisions.forEach((collision) => {
+          updateNode(collision.targetId, {
+            location: {
+              x: collision.newPosition.x,
+              y: collision.newPosition.y,
+            },
+          });
+        });
+      }
+    };
+
+    setCollisionWorker(worker);
+
+    return () => {
+      worker.terminate();
+    };
+  }, []);
+
+  const detectCollisions = useCallback(() => {
+    if (layer.current && collisionWorker) {
+      const nodes = layer.current.children
+        .filter((child) => child.attrs.name === "node")
+        .map((node) => ({
+          id: node.attrs.id,
+          location: {
+            x: node.attrs.x,
+            y: node.attrs.y,
+          },
+          rect: node.getClientRect(),
```

Comment:
사소한 거지만 여기서 필요한 정보만 추려서 보낼 수도 있을 것 같아요
```suggestion
          rect: {
          x: node.attrs.x,
          y: node.attrs.y,
          width: node.width(),
          height: node.height(),
        },
```